### ISSUE TITLE
C++ wrapper?

### ISSUE THREAD

**Issue by IngwiePhoenix** [vstakhov/libucl#6]
Hello there!

I have to say first, that this is one very great configuration library. I really like the ngix like syntax that you choose to make usable. And, this project is highly embedable as well!

But, the C-API is a little bit confusing to me still - I cant seem to find out how to iterate over an array yet? Doxygen refused to build a nice documentation too as well, so I am stuck with the test_*.c files only.

But, will there ever be a c++ wrapper? I could imagine it quite simple actually.

```
UCLObject *obj = new UCLObject(buffer);
UCLObject *srcs = obj->get("sources");
for(int i=0; i<srcs->length; i++) {
    cout << srcs->get(i)->valueAsString() << endl;
}
```

In my case, I am about ot build a meta build system using UCL. Why? The other meta build systems require a scripting language to be preinstalled. And not everyone has Python or pearl preinstalled, especially not on windows... o.o

Kind regards!
##

**Comment by davidchisnall** [comment]
Possibly too late for you, but I've been working on [some tooling](https://github.com/davidchisnall/config-gen/) for generating idiomatic C++ wrappers around libucl for config files, from JSON Schema descriptions.  These all use [a set of adaptor classes](https://github.com/davidchisnall/config-gen/blob/main/config-generic.h) that expose individual properties of UCL objects and can handle things like exposing an array as a range of `T` that you can iterate over as if it were any other C++ collection of `T`, where `T` is a wrapper around a UCL object pointer.

**Comment by vstakhov** [comment]
Yes, thank you very much! I wish I had one when generating stuff like that for Rspamd. I had also switched this project to C++ (finally) so I should be able to use your codegen tool at some point.
